plugin/evm/atomic/mempool.go

```diff
@@ -181,41 +172,36 @@ func (m *Mempool) AddTx(tx *atomic.Tx) error {
 	return err
 }
 
-func (m *Mempool) AddLocalTx(tx *atomic.Tx) error {
+func (m *Mempool) AddLocalTx(tx *Tx) error {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	err := m.addTx(tx, false)
+	err := m.addTx(tx, true, false)
 	if errors.Is(err, errTxAlreadyKnown) {
 		return nil
 	}
```

Comment:
question: What's the reason we return nil for `errTxAlreadyKnown` for local txs?

Reply:
I think it was to not return any errors to API so that it does not return error.

Reply:
I'm a little confused why we return `err` in case of `errTxAlreadyKnown` for remote txs but nil for local txs

Reply:
We can return the `err` for `AddLocalTx` but then we need to handle it in the API so that we still gossip new local tx [here](https://github.com/ava-labs/coreth/blob/move-atomic-gossip/plugin/evm/service.go#L199). I think in overall this approach is safer where re-adding known local txs assumed to be safe.